trace/amqp/integration_test.go

```diff
@@ -0,0 +1,137 @@
+// +build integration
+
+package amqp
+
+import (
+	"context"
+	"testing"
+
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPublisherSuccess(t *testing.T) {
+	ctx := context.Background()
+	pub, err := NewPublisher("amqp://guest:guest@localhost:5672/", "exchangeName")
+	assert.NoError(t, err)
+
+	originalMsg, err := NewJSONMessage(`{"status":"received"}`)
+	assert.NoError(t, err)
+	expectedMsg := `"{\"status\":\"received\"}"`
+
+	msgs := setupRabbitMQConsumer(t)
```

Comment:
you should probably also colse the consumer connections gracefully

Reply:
Done! I'm returning the collection, and deferring its Close() in place.